src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/DotNetRunner.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon
+{
+    /// <summary>
+    /// Runner for running dotnet processes.
+    /// </summary>
+    public sealed class DotNetRunner : IDisposable
+    {
+        // Event handler for the Process.Exited event
+        private readonly EventHandler _exitedHandler;
```

Comment:
Note that once you start testing other things like eventpipe, you may need communication between the two processes, similiar to: https://github.com/dotnet/diagnostics/blob/1b38833a0fb95105164cf699a04fc38c3584f548/src/tests/Microsoft.Diagnostics.NETCore.Client/RemoteTestExecution.cs#L17

Reply:
Yeah, I'll have to bring in that communication scheme once I establish a test app. I'll probably end up creating a TestAppRunner class that has this code.